### PR TITLE
FlexiblePrivacyPrecompile - Handle invalid Payload ID

### PR DESCRIPTION
**Pull request by macfarla**
Handle the case where payloadId passed in is null or < 32 bytes. 

See #3635 

Added 2 new tests for null and < 32bytes

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/precompiles/privacy/FlexiblePrivacyPrecompiledContract.java [hunk 1]
```diff
@@ -104,6 +104,10 @@ public PrecompileContractResult computePrecompile(
     if (skipContractExecution(messageFrame)) {
       return NO_RESULT;
     }
+    if (input == null || input.size() < 32) {
+      LOG.debug("Can not fetch private transaction payload with key {}", input);
```
Comment (pinges) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/precompiles/privacy/FlexiblePrivacyPrecompiledContract.java:108]: Currently we do only have two valid sizes, 32 and 64 bytes, so I guess we could do:
`if (input == null || (input.size() != 32 && input.size() != 64))`
And I think we should log that there was an invalid input and probably even log this as an error?

**Review by pinges — COMMENTED** [review]
please check out my comment

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/precompiles/privacy/FlexiblePrivacyPrecompiledContract.java (hunk 1 above)
Comment (macfarla) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/precompiles/privacy/FlexiblePrivacyPrecompiledContract.java]: done

**Review by pinges — APPROVED** [review]
LGTM

**Pull request merged**